mm2src/mm2_main/src/lp_wallet/mnemonics_storage.rs

```diff
@@ -64,8 +64,23 @@ pub(super) async fn read_encrypted_passphrase_if_available(ctx: &MmArc) -> Walle
 }
 
 pub(super) async fn read_all_wallet_names(ctx: &MmArc) -> WalletsStorageResult<impl Iterator<Item = String>> {
-    let wallet_names = list_files_by_extension(&ctx.db_root(), "dat", false)
+    let ext: &str = "dat";
+    let wallet_names = list_files_by_extension(&ctx.db_root(), ext, false)
         .await
         .mm_err(|e| WalletsStorageError::FsReadError(format!("Error reading wallets directory: {}", e)))?;
-    Ok(wallet_names)
+
+    // let's exclude directories from the wallet_names, of course we can do it on the upper levels,
+    // but let's leave `filter_files_by_extension` as is for now
+    let dir_path = ctx.db_root();
+    let mut file_wallet_names = Vec::new();
+    for wallet_name in wallet_names {
+        let mut file_path = dir_path.to_path_buf();
+        file_path.push(&wallet_name);
+        file_path.set_extension(ext);
+        if is_file_async(&file_path).await {
```

Comment:
Why not just [`is_file`](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.is_file) from std?

Reply:
Fixed.